experimental/benchmark/src/database.rs

```diff
@@ -0,0 +1,134 @@
+    #[serde(default)]
+    locked: bool,
+    #[serde(rename = "installDate", default)]
+    install_date: String,
+    #[serde(rename = "removalDate", default)]
+    removal_date: String,
+    temporary: bool,
+    #[serde(rename = "nbBikes", default)]
+    number_of_bikes: u32,
+    #[serde(rename = "nbEmptyDocks", default)]
+    number_of_empty_docks: u32,
+    #[serde(rename = "nbDocks", default)]
+    number_of_docks: u32,
+}
+
+pub struct Database {
+    pub points_of_interest: PointOfInterestMap,
+}
+
+impl Database {
+    pub fn new(size: usize) -> Self {
+        let mut rng = SmallRng::seed_from_u64(0);
+
+        let mut entries = HashMap::new();
+        for id in 0..size {
+            let name: u16 = rng.gen();
+            let latitude_degrees: f32 = rng.gen_range(-90.0, 90.0);
+            let longitude_degrees: f32 = rng.gen_range(-180.0, 180.0);
+            let entry = PointOfInterest {
+                name: format!("{}", name),
+                location: Some(Location {
+                    latitude_degrees,
+                    longitude_degrees,
+                }),
+            };
+            entries.insert(format!("{}", id), entry);
+        }
+        Self {
+            points_of_interest: PointOfInterestMap { entries },
+        }
+    }
+
+    pub fn serialize(&self) -> String {
+        let stations: Vec<Station> = self
+            .points_of_interest
+            .entries
+            .iter()
+            .map(|(key, value)| {
+                let location = value.location.as_ref().expect("Empty location");
+                Station {
+                    id: key.parse().expect("Couldn't parse key"),
+                    name: value.name.to_string(),
+                    latitude_degrees: location.latitude_degrees,
+                    longitude_degrees: location.longitude_degrees,
+                    ..Default::default()
+                }
+            })
+            .collect();
+        let station_database = StationDatabase {
+            last_update: "".to_string(),
+            version: "0.1".to_string(),
+            stations,
+        };
+        // We need to manually update the resulting string, because `quick_xml` doesn't serialize
+        // vectors correctly:
+        // https://github.com/tafia/quick-xml/issues/204
+        let database = quick_xml::se::to_string(&station_database)
+            .expect("Couldn't serialize database")
+            .replace("<station>", "")
+            .replace("</station>", "")
+            .replace("<Station ", "<station ")
+            .replace("<Station>", "<station>")
+            .replace("</Station>", "</station>");
+
+        // Measure database parsing time.
+        let start = Instant::now();
+        let parsed_database: StationDatabase =
+            quick_xml::de::from_str(&database).expect("Couldn't deserialize database");
+        let duration = start.elapsed();
+        info!(
+            "Database parsing time: {:?} for {} entries ({} bytes)",
```

Comment:
How is this information useful? AFAICT it is always done on the host natively, or am I missing something? how do you get to run this in Oak via Wasm?

Reply:
But I don't understand, isn't this code always running on the host natively? How / when does it get compiled to Wasm? Apologies if I am missing something obvious.

Reply:
Yes, it measures the native XML parsing time.
And `main.rs` has code for measuring Oak runtime loading time.